src/php/Lang/Set.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Phel\Lang;
+
+use ArrayAccess;
+use Countable;
+use InvalidArgumentException;
+use Iterator;
+use Phel\Printer;
+
+class Set extends Phel implements Countable, Iterator, ICons, IPush, IConcat
+{
+
+    /**
+     * @var mixed[]
+     */
+    protected $data = [];
+
+    /**
+     * Constructor
+     *
+     * @param mixed[] $data A list of all values
+     */
+    public function __construct(array $data)
+    {
+        $this->setData($data);
+    }
+
+    /**
+     * Create a new set.
+     *
+     * @param mixed[] $values A list of all values
+     *
+     * @return Set
+     */
+    public static function create(...$values): Set
+    {
+        return new Set($values);
+    }
+
+    public function isTruthy(): bool
+    {
+        return true;
+    }
+
+    public function hash(): string
```

Comment:
This method is unnecessary.

Reply:
It seems required since I extends AbstractType which contains the abstract method `hash`. I gave it a try and unsurprisingly it yield the fatal error `Class Phel\Lang\Set contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Phel\Lang\AbstractType::hash)`.